great_expectations/expectations/metrics/query_metric_provider.py

```diff
@@ -137,30 +100,17 @@ def _get_substituted_batch_subquery_from_query_and_batch_selectable(
 
         if isinstance(batch_selectable, sa.Table):
             query = query.format(batch=batch_selectable, **parameters)
-        elif isinstance(batch_selectable, get_sqlalchemy_subquery_type()):
-            if (
-                not parameters
-                and execution_engine.dialect_name in cls.dialect_columns_require_subquery_aliases
-            ):
-                try:
-                    query = cls._get_query_string_with_substituted_batch_parameters(
-                        query=query,
-                        batch_subquery=batch_selectable,
-                    )
-                except MissingElementError:
-                    # if we are unable to extract the subquery parameters,
-                    # we fall back to the default behavior for all dialects
-                    batch = batch_selectable.compile(compile_kwargs={"literal_binds": True})
-                    query = query.format(batch=f"({batch})", **parameters)
-            else:
-                batch = batch_selectable.compile(compile_kwargs={"literal_binds": True})
-                query = query.format(batch=f"({batch})", **parameters)
         elif isinstance(
-            batch_selectable, sa.sql.Select
-        ):  # Specifying a row_condition returns the active batch as a Select object
-            # requiring compilation & aliasing when formatting the parameterized query
+            batch_selectable, (sa.sql.Select, get_sqlalchemy_subquery_type())
+        ):  # specifying a row_condition returns the active batch as a Select
+            # specifying an unexpected_rows_query returns the active batch as a Subquery or Alias
+            # this requires compilation & aliasing when formatting the parameterized query
             batch = batch_selectable.compile(compile_kwargs={"literal_binds": True})
-            query = query.format(batch=f"({batch}) AS subselect", **parameters)
+            # all join queries require the user to have taken care of aliasing themselves
+            if "JOIN" in query.upper():
+                query = query.format(batch=f"({batch})", **parameters)
```

Comment:
The testing framework doesn't currently support `JOIN` queries because any additional tables added to the db are given a random name. This issue is being raised in timber. Regardless, a join is tested in [a docs snippet](https://github.com/great-expectations/great_expectations/blob/develop/docs/docusaurus/docs/reference/learn/data_quality_use_cases/integrity.md?plain=1#L206-L209), which is why I had to add this condition.